File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/library/items/media/files/properties/repository/IFilePropertiesContainerRepository.java
package com.lasthopesoftware.bluewater.client.library.items.media.files.properties.repository;

import com.lasthopesoftware.bluewater.client.library.items.media.files.ServiceFile;
import com.lasthopesoftware.bluewater.shared.UrlKeyHolder;

/**
 * Created by david on 3/14/17.
 */

public interface IFilePropertiesContainerRepository {
	FilePropertiesContainer getFilePropertiesContainer(UrlKeyHolder<ServiceFile> key);

	void putFilePropertiesContainer(UrlKeyHolder<ServiceFile> key, FilePropertiesContainer filePropertiesContainer);
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/library/items/media/files/properties/repository/IFilePropertiesContainerRepository.kt
package com.lasthopesoftware.bluewater.client.browsing.library.items.media.files.properties.repository

import com.lasthopesoftware.bluewater.client.browsing.library.items.media.files.ServiceFile
import com.lasthopesoftware.bluewater.shared.UrlKeyHolder

/**
 * Created by david on 3/14/17.
 */
interface IFilePropertiesContainerRepository {
	fun getFilePropertiesContainer(key: UrlKeyHolder<ServiceFile>): FilePropertiesContainer?
	fun putFilePropertiesContainer(key: UrlKeyHolder<ServiceFile>, filePropertiesContainer: FilePropertiesContainer)
}
